portal-web/test/functional/com/liferay/portalweb/macros/LiferayEvent.macro

```diff
@@ -75,4 +75,31 @@ window.LFR_JS_EVENTS_LOG = {
 		RunScript(locator1 = "${javascript}");
 	}
 
+	macro waitForLiferayEvent {
+		var javascript = '''
+try {
+	var eventValue = LFR_JS_EVENTS_LOG.getLastEvent("${eventName}").${attributeName};
+
+	if (eventValue.includes("${attributeValue}")) {
+		return "fired";
+	}
+	else {
+		return "pending";
+	}
+}
+catch(err) {
+	return "pending";
+}
+		''';
+		var liferayEventFired = "";
+
+		while ("${liferayEventFired}" != "fired" && (maxIterations = "5")) {
+			var liferayEventFired = selenium.getEval("${javascript}");
```

Comment:
Not sure what this language is, but double declaration here... and outside the while loop.

Reply:
This is Poshi syntax lol. Yeah, line 94 is unnecessary, but this will go away when we move to the function file.